src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1343 @@ ActorInit Boss_05_InitVars = {
+                           this->headJointTable, this->headMorphTable, BIO_DEKU_BABA_HEAD_LIMB_MAX);
+
+        this->dyna.actor.gravity = 0.0f;
+        this->dyna.actor.world.rot.y = Rand_ZeroFloat(0x8000);
+        this->dyna.actor.speed = Rand_ZeroFloat(3.0f) + 3.0f;
+        this->dyna.actor.velocity.y = Rand_ZeroFloat(1.5f) + 1.5f;
+
+        this->fragmentAngularVelocity.x = Rand_CenteredFloat(700.0f);
+        this->fragmentAngularVelocity.y = Rand_CenteredFloat(1500.0f);
+        this->timers[TIMER_CURRENT_ACTION] = Rand_ZeroFloat(30.0f) + 50.0f;
+
+        this->dyna.actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        this->actionFunc = Boss05_Fragment_Move;
+    }
+}
+
+void Boss05_Destroy(Actor* thisx, PlayState* play) {
+    Boss05* this = THIS;
+
+    if ((BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_LILY_PAD) ||
+        (BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_LILY_PAD_WITH_HEAD) ||
+        (BIO_BABA_GET_TYPE(&this->dyna.actor) == BIO_BABA_TYPE_NO_LEAF_LILY_PAD_WITH_HEAD)) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+/**
+ * Checks to see if the lily pad stem or the head was hit by an attack. If either of them were hit, this function will
+ * return a value indicating what the head should do; if the head should detach from the lily pad, then the draw damage
+ * effect to apply to the falling head will be included in the return value.
+ */
+s32 Boss05_LilyPadWithHead_UpdateDamage(Boss05* this, PlayState* play) {
+    if (this->damagedFlashTimer == 0) {
+        s32 i = 0;
+
+        while (true) {
+            if (this->lilyPadCollider.elements[i].info.bumperFlags & BUMP_HIT) {
+                switch (this->dyna.actor.colChkInfo.damageEffect) {
+                    case BIO_BABA_DMGEFF_FIRE:
+                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_FIRE_INIT;
+                    case BIO_BABA_DMGEFF_FREEZE:
+                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_FROZEN_INIT;
+                    case BIO_BABA_DMGEFF_LIGHT_ORB:
+                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_LIGHT_ORB_INIT;
+                    default:
+                        return BIO_BABA_HEAD_HIT_REACTION_DEATCH + BIO_BABA_DRAW_DMGEFF_STATE_NONE;
+                }
+            }
+
+            i++;
+            if (i == BIO_BABA_LILY_PAD_COLLIDER_MAX) {
+                if (this->headCollider.elements[BIO_BABA_HEAD_COLLIDER_HEAD].info.bumperFlags & BUMP_HIT) {
+                    u8 damage = this->dyna.actor.colChkInfo.damage;
+
+                    this->dyna.actor.colChkInfo.health -= damage;
+                    if ((s8)this->dyna.actor.colChkInfo.health <= 0) {
+                        Enemy_StartFinishingBlow(play, &this->dyna.actor);
+                        return BIO_BABA_HEAD_HIT_REACTION_DEAD;
+                    } else {
```

Comment:
```suggestion
                        return BIO_BABA_HEAD_HIT_REACTION_DEAD;
                    }
```
With the return, probably don't need the `else`?

Reply:
It does match without it, yes, but I dunno, I think I actually like it more with the `else`. It makes it a little more clear that this code only runs when the enemy is damaged, not killed. Though part of it might also be that I'm quite familiar with actors that do something like this, where you can't use a `return` to get ride of the `else`:
```c
if (this->actor.colChkInfo.health > 0) {
    EnJso_SetupDamaged(this, play);
} else {
    EnJso_SetupDead(this, play);
}
```
Do you have any thoughts on how this should be styled? Maybe I have a particular brand of brain poisoning from looking at too many enemy actors before this one, so I'm curious to hear what others think.

Reply:
I suppose with the else is fine. This is just personal preference, so it can go either way, but when the "else" block is has a lot more than 1 line to it, then I like just not having it if the `if` has a return. But with 1 or two lines, they look identical to me. So it's fine to leave it

Reply:
Yeah, that's a fair perspective. Let me remove it and push it up, then you can give it a look and see if you like it more.